src/main/kotlin/org/rust/ide/refactoring/extractFunction/RsExtractFunctionHandler.kt

```diff
@@ -89,11 +98,22 @@ class RsExtractFunctionHandler : RefactoringActionHandler {
             }
     }
 
-    private fun createNewInherentImpl(traitImpl: RsImplItem): RsImplItem {
+    private fun createNewInherentImpl(traitImpl: RsImplItem): RsImplItem? {
         val parent = traitImpl.parent
         val psiFactory = RsPsiFactory(parent.project)
-        val type = traitImpl.typeReference!!
-        val newImpl = psiFactory.createImpl(type.text, emptyList())
+
+        val typeRef = traitImpl.typeReference!!
+        val constraints = GenericConstraints.create(traitImpl).filterByTypes(listOf(typeRef.type))
```

Comment:
Looks like it doesn't work with lifetimes:
```rust
struct Foo<'a> { t: &'a i32 }
impl<'a> Trait for Foo<'a> {
...
```
```rust
trait Trait2<'a> {}
struct Foo<T> { t: T }
impl<'a, T> Trait for Foo<T> where T: Trait2<'a> {
...
```
Consider adjusting [`gatherLifetimesFromTypeParameters`](https://github.com/intellij-rust/intellij-rust/blob/514ffd677a8e21bb4d9e80ddee75f9d541b00db1/src/main/kotlin/org/rust/ide/utils/ExtractUtils.kt#L53)

Reply:
Thanks. The filtering should have been done by the type reference, not by its type. I changed it and added a test with lifetimes.